pkg/core/transactor/transactor.go

```diff
@@ -105,6 +115,27 @@ func (t *Transactor) Listen() {
 	}
 }
 
+func (t *Transactor) canStake() bool {
+	if t.getSyncProgress() == 100 {
+		return true
+	}
+
+	// Check for our sync progress three more times. If we still can't stake after
+	// the third check, it's better to just return false, and give the Transactor
+	// control back over the Listen goroutine.
+	interval := 5 * time.Second
+
+	for i := 0; i < 3; i++ {
```

Comment:
When I suggested to have a retry mechanism, I meant that it could be useful at the grpc client-side mostly (e.g wallet cli) where retry period can be also configurable and we will not block the entire Transactor on each stake Tx. But I believe server-side mechanism could be good for now too.

Reply:
The reason I opted for a server-side mechanism is because the grpc client is not the only one that can ask the transactor to stake (the `StakeAutomaton` does too). But, your point does remind me that the retry is therefore in the wrong place, as it needs to be touched by both code paths.